### PR TITLE
Fixed codecs to mimetype conversion to take into account all possible scenarios

### PR DESCRIPTION
**Pull request by imbcmdth**
## Description
There are 4 possible predicates that influence how to generate mimetypes from codec strings:

* Video codec specified
* Audio codec specified
* Are there multiple audio tracks (audio media-groups)
* If so, is the primary media demuxed (lacking audio) 

This PR adds complete unit testing of all 24 possible combinations and fixes issues around the function that caused audio-only streams with multiple audio tracks present.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Unit Tests updated or fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: test/master-playlist-controller.test.js [hunk 1]
```diff
@@ -19,6 +19,40 @@ import { Hls } from '../src/videojs-contrib-hls';
 /* eslint-enable no-unused-vars */
 import Playlist from '../src/playlist';
 
+const generateMedia = function(isMaat, isMuxed, hasVideoCodec, hasAudioCodec) {
+  const codec = (hasVideoCodec ? 'avc1.4d400d' : '') +
```
Comment (gesinger) [inline test/master-playlist-controller.test.js]: Would be good to use different codecs from the defaults when testing provided codecs

**Comment by gesinger** [comment]
We should also have tests for `video/mp4` and `audio/mp4`

File: test/master-playlist-controller.test.js (hunk 1 above)
Comment (imbcmdth) [inline test/master-playlist-controller.test.js]: Good point.

File: src/master-playlist-controller.js [hunk 2]
```diff
@@ -14,6 +14,13 @@ import Decrypter from './decrypter-worker';
 
 let Hls;
 
+// Default codec parameters if non were provided for video and/or audio
+const defaultCodecs = {
+  videoCodec: 'avc1',
+  videoObjectTypeIndicator: '.4d400d',
```
Comment (imbcmdth) [inline src/master-playlist-controller.js]: I'm not sure there is a reason other than "This was a common configuration when we started writing this project..." which is a crappy comment :1st_place_medal:

File: src/master-playlist-controller.js [hunk 3]
```diff
@@ -14,6 +14,13 @@ import Decrypter from './decrypter-worker';
 
 let Hls;
 
+// Default codec parameters if non were provided for video and/or audio
```
Comment (imbcmdth) [inline src/master-playlist-controller.js]: Done

File: src/master-playlist-controller.js [hunk 4]
```diff
@@ -119,74 +159,87 @@ export const mapLegacyAvcCodecs_ = function(codecString) {
...
 
-  // if the codecs were explicitly specified, use them instead of the
-  // defaults
-  mediaAttributes = media.attributes || {};
-  if (mediaAttributes.CODECS) {
-    let parsedCodecs = parseCodecs(mediaAttributes.CODECS);
+  // HLS with multiple-audio tracks must always get an audio codec.
+  // Put another way, there is no way to have a video-only multiple-audio HLS!
+  if (isMaat && !codecInfo.audioProfile) {
+    codecInfo.audioProfile = defaultCodecs.audioProfile;
```
Comment (imbcmdth) [inline src/master-playlist-controller.js]: I don't know.. It seems like a fairly common configuration with MAAT unfortunately.. I add a warning log.

File: test/master-playlist-controller.test.js [hunk 5]
```diff
@@ -1589,115 +1633,87 @@ QUnit.test('sets up subtitles', function(assert) {
...
+
+  assert.deepEqual(mimeTypesForPlaylist_.apply(null,
+      generateMedia(true, false, true, false, isFMP4)),
+    [`${videoMime}; codecs="avc1.deadbeef"`,
+     `${audioMime}; codecs="mp4a.40.2"`],
+    `MAAT, demuxed, container: ${container}, codecs: video`);
+
+  assert.deepEqual(mimeTypesForPlaylist_.apply(null,
+      generateMedia(true, false, false, true, isFMP4)),
+    [`${videoMime}; codecs="mp4a.40.E"`,
```
Comment (gesinger) [inline test/master-playlist-controller.test.js]: This still seems so odd to me.

File: src/master-playlist-controller.js [hunk 6]
```diff
@@ -119,74 +171,92 @@ export const mapLegacyAvcCodecs_ = function(codecString) {
...
+    return [
+      bothVideoAudio,
+      justAudio
+    ];
   }
-  // if all video and audio is unmuxed, use two single-codec mime
-  // types
-  if (previousGroup && previousGroup.uri) {
+
+  // If there is ano video codec at all, always just return a single
```
Comment (gesinger) [inline src/master-playlist-controller.js]: ano

**Pull request merged**